### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by drothlis**
- [x] Fix `os.environ` keys and/or values can't be unicode on Python 2 on Windows.
- [x] Use requirements.txt in docker scripts too

### CODE REVIEW

File: stbt_rig.py [hunk 1]
```diff
@@ -1149,5 +1152,33 @@ def die(message, *args):
...
+            raise exc.with_traceback(tb)
+        raise exc
+else:
+    # `raise a, b, c` is a syntax error on Python 3 (even though we don't run
+    # this block with Python 3, Python still has to parse it). Hence `exec`.
+    exec(  # pylint:disable=exec-used
+        dedent('''\
+        def raise_(tp, value=None, tb=None):
+            raise tp, value, tb
+        '''))
```
Comment (wmanley) [inline stbt_rig.py:1201]: Clever

**Comment by drothlis** [comment]
`test_run_tests_jenkins` hangs on the AppVeyor (Windows) CI, but then it has always hung since I added the test in #23. I guess I need set up a Windows VM to investigate properly.

**Comment by drothlis** [comment]
I have also tested d525641 manually in interactive & pytest modes with Python 2.7 & 3.6, on Linux.
I'm tempted to merge and fix the Windows tests on a separate pull request (`test_run_tests_jenkins` has always hung on Windows, since the test was introduced).

**Comment by drothlis** [comment]
Ah no, can't merge yet -- I've broken `os.environ` keys and/or values can't be unicode on Python 2 on Windows.

File: stbt_rig.py [hunk 2]
```diff
@@ -882,19 +882,14 @@ def __init__(self, root=None, remote="origin"):
         self.root = root
         self.remote = remote
 
-    def _git(self, cmd, capture_output=True, extra_env=None, **kwargs):  # pylint:disable=no-self-use
-        if capture_output:
-            call = subprocess.check_output
-        else:
-            call = subprocess.check_call
-
+    def _git(self, cmd, extra_env=None, **kwargs):  # pylint:disable=no-self-use
```
Comment (wmanley) [inline stbt_rig.py:885]: > Refactor: Remove unused `capture_output` argument 

I think this should be set to `False` for our call to git push during the snapshot (at least in interactive mode)?  This way people can enter passwords.

Comment (wmanley) [reply stbt_rig.py:885]: Of course this isn't a regression so let's not worry about it for now. It can be reintroduced in another PR.

**Comment by wmanley** [comment]
> I'm tempted to merge and fix the Windows tests on a separate pull request (test_run_tests_jenkins has always hung on Windows, since the test was introduced).

I'm happy with this.  I'll merge if you're happy with the changes I made.

**Pull request merged**